### PR TITLE
Add tram stop by stop

### PR DESCRIPTION
**Pull request by gldgrnt**
(GeoJSON stuff not included)

### CODE REVIEW

File: src/components/shared/Tray/TrayComponents/AutoComplete/TrainAutoComplete/TrainAutocomplete.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ const TrainAutoComplete = ({ to }) => {
             </div>
             <DebounceInput
               type="text"
-              name="tramSearch"
+              name="trainSearch"
               placeholder="Search for a stop"
```
Comment (daylesalmon) [inline src/components/shared/Tray/TrayComponents/AutoComplete/TrainAutoComplete/TrainAutocomplete.js]: Can we change this to "Search for a station"

File: src/components/shared/Tray/TrayComponents/SelectedService/InfoAboutSelectedService/TramInfoAboutSelectedService/TramInfoAboutSelectedService.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import FavBtn from 'components/shared/FavBtn/FavBtn';
+import Message from 'components/shared/Message/Message';
+// Local
+import numberToWord from '../helpers/numberToWord';
+import useGetTramStopByStop from './customHooks/useGetTramStopByStop';
+
+const TramInfoAboutSelectedService = () => {
+  const [autoCompleteState] = useContext(AutoCompleteContext);
+  const { selectedItem, selectedItemTo } = autoCompleteState;
+  const { loading, errorInfo } = useGetTramStopByStop();
```
Comment (daylesalmon) [inline src/components/shared/Tray/TrayComponents/SelectedService/InfoAboutSelectedService/TramInfoAboutSelectedService/TramInfoAboutSelectedService.js]: To tidy this file a bit more, return the autoCompleteState from the useGetTramStopByStop.

File: src/components/shared/Tray/TrayComponents/SelectedService/InfoAboutSelectedService/TramInfoAboutSelectedService/customHooks/useGetTramStopByStop.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import { useState, useEffect, useContext } from 'react';
+import axios from 'axios';
+import { AutoCompleteContext } from 'globalState';
+
+const useGetTramStopByStop = () => {
+  const [autoCompleteState, autoCompleteDispatch] = useContext(AutoCompleteContext);
+  const { selectedItem, selectedItemTo } = autoCompleteState;
+  // Intitialise state
```
Comment (daylesalmon) [inline src/components/shared/Tray/TrayComponents/SelectedService/InfoAboutSelectedService/TramInfoAboutSelectedService/customHooks/useGetTramStopByStop.js]:
```suggestion
  // initialise state
```

File: src/components/shared/Tray/TrayComponents/SelectedService/InfoAboutSelectedService/TramInfoAboutSelectedService/customHooks/useGetTramStopByStop.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            title: 'Please try again',
+            message: 'Apologies, we are having technical difficulties.',
+          });
+          // eslint-disable-next-line no-console
+          console.log({ error });
+        }
+      });
+  }, [autoCompleteDispatch, bothStopsEqual, bothStopsSelected, selectedItem.id, selectedItemTo.id]);
+
+  return { loading, errorInfo };
```
Comment (daylesalmon) [inline src/components/shared/Tray/TrayComponents/SelectedService/InfoAboutSelectedService/TramInfoAboutSelectedService/customHooks/useGetTramStopByStop.js]: Return the autoCompleteState here so you can just use it in your other file.

File: src/components/shared/Tray/TrayComponents/SelectedService/InfoAboutSelectedService/helpers/numberToWord.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    'Twenty-one',
+    'Twenty-two',
+    'Twenty-three',
+    'Twenty-four',
+    'Twenty-five',
+    'Twenty-six',
+    'Twenty-seven',
+    'Twenty-eight',
+    'Twenty-nine',
+    'Thirty',
```
Comment (daylesalmon) [inline src/components/shared/Tray/TrayComponents/SelectedService/InfoAboutSelectedService/helpers/numberToWord.js:34]: Check all the typos in here. Some "Thirty" are "Thrity".

File: src/customHooks/useFilterLogic.js [hunk 6]
```diff
@@ -60,9 +60,15 @@ const useFilterLogic = () => {
       switch (modeState.mode) {
         // The mode is tram and the id the user clicked in the autocomplete is within the stopsAffected array
         case 'tram':
-          filteredData = filteredData.filter((disrItem) =>
-            disrItem.stopsAffected.some((el) => el.atcoCode === autoCompleteState.selectedItem.id)
-          );
+          // If the user has selected both tram stops then check if any are affected
+          if (autoCompleteState.selectedItemTo.id && autoCompleteState.selectedItem?.lines) {
+            // Map the array of stop objects to an array of just the atcoCodes
+            const lineCodes = autoCompleteState.selectedItem.lines.map((stop) => stop.atcoCode);
```
Comment (daylesalmon) [inline src/customHooks/useFilterLogic.js:66]: Lets add "atcoCode" to the typo.yml exclude list.

**Review by daylesalmon — CHANGES_REQUESTED** [review]
Just  a few changes.

Another bug I have noticed is...

- Fresh start the app
- Click tram
- Select any two stops
- The API fetches the in-between stops
- Whilst this fetching is going on, the spinner shows but also "good service" remains.

Whilst fetching, the good service shouldn't show until the API has resolved.

**Comment by gldgrnt** [comment]
> Just a few changes.
> 
> Another bug I have noticed is...
> 
> * Fresh start the app
> * Click tram
> * Select any two stops
> * The API fetches the in-between stops
> * Whilst this fetching is going on, the spinner shows but also "good service" remains.
> 
> Whilst fetching, the good service shouldn't show until the API has resolved.

@daylesalmon I'm fixing this in another branch, where I'm also fixing the bug where the 'Good Service' message should show straight away for modes with no disruptions. I'll create a PR once this has been merged 👍🏾

**Review by daylesalmon — APPROVED** [review]
All good!

**Pull request merged**